packages/angular/src/migrations/update-14-5-0/migrate-mfe-to-mf.ts

```diff
@@ -0,0 +1,90 @@
+import type { Tree } from '@nrwl/devkit';
+import { visitNotIgnoredFiles } from '@nrwl/devkit';
+import { extname } from 'path';
+import { tsquery } from '@phenomnomnominal/tsquery';
+import { SourceFile } from 'typescript';
+
+const NRWL_ANGULAR_MFE_STATIC_IMPORT_SELECTOR =
+  'ImportDeclaration > StringLiteral[value="@nrwl/angular/mfe"]';
+
+const NRWL_ANGULAR_MFE_DYNAMIC_IMPORT_SELECTOR =
+  'CallExpression:has(ImportKeyword) > StringLiteral[value="@nrwl/angular/mfe"]';
+
+const NRWL_ANGULAR_MFE_TYPES_SELECTOR =
+  'ImportDeclaration:has(StringLiteral[value=@nrwl/angular/module-federation]) > ImportClause > NamedImports';
+
+export function replaceNrwlAngularMfImport(fileContents: string) {
+  let fileAst: SourceFile = tsquery.ast(fileContents);
+  if (fileContents.includes('@nrwl/angular/mfe')) {
+    // This file definitely contains the string, however, we're interested in whether it is an import
+    const staticQueryResult = tsquery(
+      fileAst,
+      NRWL_ANGULAR_MFE_STATIC_IMPORT_SELECTOR,
+      {
+        visitAllChildren: true,
+      }
+    );
+
+    if (staticQueryResult && staticQueryResult.length > 0) {
+      fileContents = `${fileContents.slice(
+        0,
+        staticQueryResult[0].getStart()
+      )}'@nrwl/angular/mf'${fileContents.slice(staticQueryResult[0].getEnd())}`;
+    }
+
+    fileAst = tsquery.ast(fileContents);
+
+    const dynamicQueryResult = tsquery(
+      fileAst,
+      NRWL_ANGULAR_MFE_DYNAMIC_IMPORT_SELECTOR,
+      {
+        visitAllChildren: true,
+      }
+    );
+
+    if (dynamicQueryResult && dynamicQueryResult.length > 0) {
+      fileContents = `${fileContents.slice(
+        0,
+        dynamicQueryResult[0].getStart()
+      )}'@nrwl/angular/mf'${fileContents.slice(
+        dynamicQueryResult[0].getEnd()
+      )}`;
+    }
+  }
+  return fileContents;
+}
+
+export function replaceExportedMFETypes(fileContents: string) {
+  const ast = tsquery.ast(fileContents);
+  const queryResult = tsquery(ast, NRWL_ANGULAR_MFE_TYPES_SELECTOR, {
+    visitAllChildren: true,
+  });
+
+  if (queryResult && queryResult.length > 0) {
+    const TYPES_IMPORTED_FROM_NRWL_REGEX =
+      /(MFERemotes|MFEConfig)+.*from+.+(@nrwl\/angular\/module-federation)+/g;
+    if (TYPES_IMPORTED_FROM_NRWL_REGEX.test(fileContents)) {
+      fileContents = fileContents.replace(/MFERemotes/g, 'MFRemotes');
+      fileContents = fileContents.replace(/MFEConfig/g, 'MFConfig');
+    }
+  }
+  return fileContents;
+}
+
+export default async function (tree: Tree) {
```

Comment:
One thing this is missing is looking for code using the generators that were changed. There can be folks creating their own generators by composing our own.